src/gnu/trove/array/AbstractOffheapArray.java

```diff
@@ -90,8 +127,38 @@ public void free() {
 
         // Cleaner is guaranteed to run its runnable at most once, so its fine to
         // call free multiple times, and nothing additional will happen when this
-        // array becomes unreachable.
-        cleaner.clean();
+        // array becomes unreachable.`
+        clean();
+    }
+
+    private void clean() {
+        try {
+            Class<?> cleanerClass = Class.forName("sun.misc.Cleaner");
+            Method clean = cleanerClass.getMethod("clean");
+            clean.invoke(cleaner);
```

Comment:
Do we want to return early in the successful case?

Reply:
Yeah, I'll change that